### PR TITLE
feat(oidc): end session by id_token_hint and without cookie

### LINKED ISSUE #8501: OIDC: allow end_session_endpoint requests without user agent cookie
The end_session_endpoint currently always requires the userAgent cookie to be able to terminate session created through the hosted login UI.
Only tokens issued through the Login V2 can be used to directly terminate a specific session and without the need of a cookie.

By adding the `sid` claim for the login UI session (https://github.com/zitadel/zitadel/issues/8499), we'll be able to determine which session to terminate even without the user agent cookie.

To avoid breaking behavior changes, we should add a feature flag to allow the termination of a single "v1 session" when sending an id_token_hint with a "v1" sid.

## Acceptance criteria
- [x] A feature flag can be set to allow single session termination for v1 sessions by id_token_hint
- [x] Terminate the v1 session if an id_token_hint is sent with a `sid` (without v2 prefix) and the flag is active
- [x] Terminate all v1 sessions of a user agent if an id_token_hint is sent with a `sid` (without v2 prefix) and the flag is inactive
- [x] Terminate all v1 sessions of a user agent if no id_token_hint is sent (still identify user agent by cookie; as is)

### PR DESCRIPTION
**Pull request by livio-a**
# Which Problems Are Solved

The end_session_endpoint currently always requires the userAgent cookie to be able to terminate session created through the hosted login UI.
Only tokens issued through the Login V2 can be used to directly terminate a specific session and without the need of a cookie.
This PR adds the possibility to terminate a single V1 session or all V1 sessions belonging to the same user agent without the need of the userAgent cookie by providing an id_token as `id_token_hint` which contains the id of a V1 session as `sid`.

# How the Problems Are Solved

- #8525 added the `sid` claim for id_tokens issued through the login UI
- The `sid` can now be checked for the `V1_` prefix and queries for either the userAgentID and depending on the `OIDCSingleV1SessionTermination` flag all userIDs of active session from the same user agent id
- The `OIDCSingleV1SessionTermination` flag is added with default value false to keep the existing behavior of terminating all sessions even in case of providing an id_token_hint

# Additional Changes

- pass `context.Context` into session view functions for querying the database with that context

# Additional Context

- relates to #8499 
- closes #8501

### CODE REVIEW

File: internal/api/grpc/feature/v2beta/converter.go [hunk 1]
```diff
@@ -17,6 +17,7 @@ func systemFeaturesToCommand(req *feature_pb.SetSystemFeaturesRequest) *command.
 		Actions:                         req.Actions,
 		TokenExchange:                   req.OidcTokenExchange,
 		ImprovedPerformance:             improvedPerformanceListToDomain(req.ImprovedPerformance),
+		OIDCSingleV1SessionTermination:  req.OidcSingleV1SessionTermination,
```
Comment (stebenz) [inline internal/api/grpc/feature/v2beta/converter.go:20]: Do we need to update v2beta as well as v2?

File: internal/user/repository/view/active_user_ids_by_session_id.sql [hunk 2]
```diff
@@ -0,0 +1,11 @@
+SELECT
+    s.user_agent_id,
+    s.user_id
+FROM auth.user_sessions s
+    JOIN auth.user_sessions s2
```
Comment (stebenz) [inline internal/user/repository/view/active_user_ids_by_session_id.sql:5]: Why do you do a JOIN here?

File: internal/api/grpc/feature/v2beta/converter.go (hunk 1 above)
Comment (livio-a) [reply internal/api/grpc/feature/v2beta/converter.go:20]: 🤷 we did so with previous feature flags

File: internal/user/repository/view/active_user_ids_by_session_id.sql (hunk 2 above)
Comment (livio-a) [reply internal/user/repository/view/active_user_ids_by_session_id.sql:5]: i need to either self join or sub query for the `user_agent_id` of the passed `id` to get all `user_id`s

**Review by peintnermax — APPROVED** [review]

**Pull request merged**